Sanitizer.cmake

```diff
@@ -0,0 +1,48 @@
+# Used to turn on certain sanitizers, if available:
+# 
+# clang:
+#    -dumpcompiler: 4.2.1
+#       address
+#       thread
+#       undefined
+# gcc:
+#    -dumpcompiler: 4.8.3
+#       address
+#       thread
+#
+# Note: This modifies CMAKE_CXX_FLAGS and CMAKE_EXE_LINKER_FLAGS
+
+option(SANITIZER "Sanitizer: one of address/thread/undefined, depending on compiler" OFF)
+if(NOT SANITIZER)
+  return()
+endif()
+
+#According to: http://clang.llvm.org/docs/AddressSanitizer.html#usage
+# the following compiler flags allow us to keep -O1 on, if desired, but
+# still get pretty call stacks
+set(_SANITIZER_COMPILE_OPTIONS "-g -fno-omit-frame-pointer -fno-optimize-sibling-calls")
+
+string(TOLOWER ${SANITIZER} SANITIZER)
+
+if(CMAKE_COMPILER_IS_GNUCXX_PURE)
```

Comment:
I'm expecting that to already happen, like in Findcppcheck.cmake and Coverage.cmake, is that not a reasonable assumption?